tests/kernel/timer/timer_api/src/main.c

```diff
@@ -544,7 +556,19 @@ void test_timer_remaining(void)
 	zassert_true(rem_ms <= (DURATION / 2) + k_ticks_to_ms_floor64(1),
 		     NULL);
 
-	zassert_true(rem_ticks <= ((dur_ticks / 2) + 1), NULL);
+	/* Half the value of DURATION in ticks may not be the value of
+	 * half DURATION in ticks, when DURATION/2 is not an integer
+	 * multiple of ticks, so target_rem_ticks is used rather than
+	 * dur_ticks/2.
+	 *
+	 * Also if the tick clock is faster or slower than the
+	 * busywait clock the remaining time in ticks will be larger
+	 * or smaller than expected, so relax the tolerance.  3 ticks
+	 * variance has been observed on hardware where the busywait
+	 * clock is 0.7% slow and the 32 KiHz tick clock is 60 ppm
+	 * fast relative to a stable external clock.
```

Comment:
Hm... what this does is basically change the Zephyr tested promise that the busy wait and system clock times are compatible.  I think if we're going to do this as policy it needs to be spelled out in the docs explicitly somewhere, not just silently waved away in a test.

Reply:
Done.